## EDA sanity checking plotting and checking missing values 
- Checking 9/11 

In [1]:
import pandas as pd
%matplotlib inline

In [2]:
df = pd.read_json("data/silver/1920_2024.json")

In [3]:
df.Aboard[0]

{'total': 35, 'passengers': 31, 'crew': 4}

In [4]:
# Checking 9/11
df[df.Date == "2001-09-11"]

Decade       Date  Time                   Location           Operator  \
2757    2000 2001-09-11  0945       Arlington, Virginia.  American Airlines   
2758    2000 2001-09-11  0847    New York City, New York  American Airlines   
2759    2000 2001-09-11  0903    New York City, New York   United Air Lines   
2760    2000 2001-09-11  1003  Shanksville, Pennsylvania   United Air Lines   

     Flight_Num                          Route           Ac_Type Registration  \
2757         77  Washington D.C. - Los Angeles  Boeing B-757-223       N644AA   
2758         11           Boston - Los Angeles  Boeing 767-223ER       N334AA   
2759        175           Boston - Los Angeles  Boeing B-767-222       N612UA   
2760         93         Newark - San Francisco  Boeing B-757-222       N591UA   

          Cn_Ln                                       Aboard  \
2757  24602/365   {'total': 64, 'passengers': 58, 'crew': 6}   
2758  22332/169  {'total': 92, 'passengers': 81, 'crew': 11}   
2759   21873/41   {'total': 65, 'passengers': 56, 'crew': 9}   
2760  28142/718   {'total': 44, 'passengers': 37, 'crew': 7}   

                                       Fatalities  Ground  \
2757   {'total': 64, 'passengers': 58, 'crew': 6}   125.0   
2758  {'total': 92, 'passengers': 81, 'crew': 11}  2750.0   
2759   {'total': 65, 'passengers': 56, 'crew': 9}  2750.0   
2760   {'total': 44, 'passengers': 37, 'crew': 7}     0.0   

                                                Summary  Aboard_Total  \
2757  The aircraft was hijacked after taking off fro...            64   
2758  The aircraft was hijacked shortly after it lef...            92   
2759  The aircraft was hijacked shortly after it lef...            65   
2760  The aircraft was hijacked after taking off fro...            44   

      Fatalities_Total Operator_vs_Military  
2757                64    American Airlines  
2758                92    American Airlines  
2759                65     United Air Lines  
2760                44     United Air Lines

For the two plains in 9/11 the ground was counted twice
- We split the count into two 2750 / 2 

In [5]:
print(df.iloc[2759].Ground / 2)
df.loc[2758, "Ground"] = 1375
df.loc[2759, "Ground"] = 1375

0.0


Compare Data source with wiki to cross reference dates and fatalities 

In [6]:
df[df.Date == "2001-09-11"]

Decade       Date  Time                   Location           Operator  \
2757    2000 2001-09-11  0945       Arlington, Virginia.  American Airlines   
2758    2000 2001-09-11  0847    New York City, New York  American Airlines   
2759    2000 2001-09-11  0903    New York City, New York   United Air Lines   
2760    2000 2001-09-11  1003  Shanksville, Pennsylvania   United Air Lines   

     Flight_Num                          Route           Ac_Type Registration  \
2757         77  Washington D.C. - Los Angeles  Boeing B-757-223       N644AA   
2758         11           Boston - Los Angeles  Boeing 767-223ER       N334AA   
2759        175           Boston - Los Angeles  Boeing B-767-222       N612UA   
2760         93         Newark - San Francisco  Boeing B-757-222       N591UA   

          Cn_Ln                                       Aboard  \
2757  24602/365   {'total': 64, 'passengers': 58, 'crew': 6}   
2758  22332/169  {'total': 92, 'passengers': 81, 'crew': 11}   
2759   21873/41   {'total': 65, 'passengers': 56, 'crew': 9}   
2760  28142/718   {'total': 44, 'passengers': 37, 'crew': 7}   

                                       Fatalities  Ground  \
2757   {'total': 64, 'passengers': 58, 'crew': 6}   125.0   
2758  {'total': 92, 'passengers': 81, 'crew': 11}  1375.0   
2759   {'total': 65, 'passengers': 56, 'crew': 9}  1375.0   
2760   {'total': 44, 'passengers': 37, 'crew': 7}     0.0   

                                                Summary  Aboard_Total  \
2757  The aircraft was hijacked after taking off fro...            64   
2758  The aircraft was hijacked shortly after it lef...            92   
2759  The aircraft was hijacked shortly after it lef...            65   
2760  The aircraft was hijacked after taking off fro...            44   

      Fatalities_Total Operator_vs_Military  
2757                64    American Airlines  
2758                92    American Airlines  
2759                65     United Air Lines  
2760                44     United Air Lines

In [7]:
df[df.Date == "1977-03-27"]

Decade       Date  Time                  Location  \
4639    1970 1977-03-27  1707  Tenerife, Canary Islands   

                              Operator Flight_Num  \
4639  Pan American World Airways / KLM  1736/4805   

                                              Route  \
4639  Tenerife - Las Palmas / Tenerife - Las Palmas   

                                   Ac_Type   Registration  \
4639  Boeing B-747-121 / Boeing B-747-206B  N736PA/PH-BUF   

                     Cn_Ln                                         Aboard  \
4639  19643/11 / 20400/157  {'total': 644, 'passengers': 614, 'crew': 30}   

                                         Fatalities  Ground  \
4639  {'total': 583, 'passengers': 560, 'crew': 23}     0.0   

                                                Summary  Aboard_Total  \
4639  Both aircraft were diverted to Tenerife becaus...           644   

      Fatalities_Total              Operator_vs_Military  
4639               583  Pan American World Airways / KLM

In [8]:
df.Date.value_counts()[df.Date.value_counts() > 1]
# Good to know that we can have more than one crash in a single day

Date
2001-09-11    4
1988-08-31    4
1962-11-23    3
1970-12-31    3
1947-05-29    3
             ..
1966-01-01    2
1947-10-16    2
2001-10-10    2
1973-09-30    2
1998-01-13    2
Name: count, Length: 398, dtype: int64

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 5020 entries, 0 to 5036
Data columns (total 17 columns):
 #   Column                Non-Null Count  Dtype         
---  ------                --------------  -----         
 0   Decade                5020 non-null   int64         
 1   Date                  5020 non-null   datetime64[ns]
 2   Time                  3528 non-null   object        
 3   Location              5015 non-null   object        
 4   Operator              5009 non-null   object        
 5   Flight_Num            1340 non-null   object        
 6   Route                 4270 non-null   object        
 7   Ac_Type               5007 non-null   object        
 8   Registration          4754 non-null   object        
 9   Cn_Ln                 4364 non-null   object        
 10  Aboard                5020 non-null   object        
 11  Fatalities            5020 non-null   object        
 12  Ground                4988 non-null   float64       
 13  Summary               4